tests/consensus_tests/test_partial_snapshot.py

```diff
@@ -31,6 +32,34 @@ def test_partial_snapshot(tmp_path: pathlib.Path, bootstrap_points: int, recover
     recover_partial_snapshot_from(read_peer, write_peer)
     assert_consistency(read_peer, write_peer)
 
+def test_partial_snapshot_recovery_lock(tmp_path: pathlib.Path):
+    assert_project_root()
+
+    write_peer, read_peer = bootstrap_peers(tmp_path, 100_000)
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers = 3)
+    futures = [executor.submit(try_recover_partial_snapshot_from, read_peer, write_peer, wait = False) for _ in range(3)]
+    responses = [future.result() for future in concurrent.futures.as_completed(futures)]
+
+    assert any(response.status_code == 400 for response in responses), "Subsequent partial snapshot recovery requests have to be rejected during partial snapshot recovery"
+
+def test_partial_snapshot_read_lock(tmp_path: pathlib.Path):
+    assert_project_root()
+
+    write_peer, read_peer = bootstrap_peers(tmp_path, 100_000)
+
+    executor = concurrent.futures.ThreadPoolExecutor(max_workers = 1)
+    recover_future = executor.submit(recover_partial_snapshot_from, read_peer, write_peer)
+
+    is_search_rejected = False
+    while not recover_future.done():
+        response = try_search_random(read_peer)
+        if response.status_code == 500:
+            is_search_rejected = True
+            break
+
+    assert is_search_rejected, "Search requests have to be rejected during partial snapshot recovery"
```

Comment:
Affecting search availability might be not be a good idea? Although, I agree that we need to take it down to load the new temp directory snapshot :thinking:

Reply:
That's by design. We want to reject read requests while local replica is unavailable due to partial snapshot being recovered.